chart/templates/operator.yaml

```diff
@@ -288,40 +287,54 @@ spec:
       labels:
         app: status-controller
     spec:
+      securityContext:
+        runAsNonRoot: true
+      serviceAccountName: status-sa
+      initContainers:
+      - name: create-crd
+        image: quay.io/kubestellar/kubectl:1.29.3
+        command: ['sh', '-c']
+        args:
+        - |
+          echo -n "Waiting for managedclusteraddons CRD"
+          while [ "$(kubectl get crd clustermanagementaddons.addon.open-cluster-management.io 2> /dev/null)" == "" ] ; do
```

Comment:
FYI, I think the test here could be simpler. We do not really care what defines the resource.

```
while ! kubectl get clustermanagementaddons.addon.open-cluster-management.io &> /dev/null ; do
```

Reply:
ok, thank you for the good suggestion, I have used your implementation

Reply:
Also, FYI: looking for the existence of the CRD object is not enough. When a CRD object is first created, it takes the apiserver a while to process that definition and fully put it into effect. This script would have to wait for that too. It is indicated by a Condition in the CRD's status, with this type: https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1/types.go#L305-L308